cdb2jdbc/src/main/java/com/bloomberg/comdb2/jdbc/Comdb2Handle.java

```diff
@@ -1532,7 +1533,17 @@ else if (!isHASqlCommit && (is_commit || is_rollback)) {
     }
 
     @Override
-    public synchronized int runStatement(String sql, List<Integer> types) {
+    public int runStatement(String sql, List<Integer> types) {
+        lock.lock();
```

Comment:
It's a very slight change in behaviour. Now `runStatement`, `next` and `close` use the same lock for the synchronisation. Previously `close` could run in parallel to the other two methods.